miku-server/src/main/scala/org/http4s/miku/MikuBuilder.scala

```diff
@@ -0,0 +1,492 @@
+    maxHeaderSize: Int,
+    maxChunkSize: Int,
+    sslBits: Option[MikuSSLConfig],
+    serviceErrorHandler: ServiceErrorHandler[F],
+    transport: NettyTransport,
+    ec: ExecutionContext,
+    enableWebsockets: Boolean,
+    banner: immutable.Seq[String],
+    nettyChannelOptions: MikuBuilder.NettyChannelOptions
+)(implicit F: Effect[F])
+    extends ServerBuilder[F]
+    with IdleTimeoutSupport[F]
+    with SSLKeyStoreSupport[F]
+    with SSLContextSupport[F]
+    with WebSocketSupport[F] {
+  implicit val e = ec
+  private val logger = getLogger
+
+  type Self = MikuBuilder[F]
+
+  def copy(
+      httpService: HttpService[F] = httpService,
+      socketAddress: InetSocketAddress = socketAddress,
+      idleTimeout: Duration = Duration.Inf,
+      maxInitialLineLength: Int = maxInitialLineLength,
+      maxHeaderSize: Int = maxHeaderSize,
+      maxChunkSize: Int = maxChunkSize,
+      sslBits: Option[MikuSSLConfig] = sslBits,
+      serviceErrorHandler: ServiceErrorHandler[F] = serviceErrorHandler,
+      ec: ExecutionContext = ec,
+      enableWebsockets: Boolean = enableWebsockets,
+      banner: immutable.Seq[String] = banner,
+      transport: NettyTransport = transport,
+      nettyChannelOptions: MikuBuilder.NettyChannelOptions = nettyChannelOptions
+  ): MikuBuilder[F] =
+    new MikuBuilder[F](
+      httpService,
+      socketAddress,
+      idleTimeout,
+      maxInitialLineLength,
+      maxHeaderSize,
+      maxChunkSize,
+      sslBits,
+      serviceErrorHandler,
+      transport,
+      ec,
+      enableWebsockets,
+      banner,
+      nettyChannelOptions
+    )
+
+  def bindSocketAddress(socketAddress: InetSocketAddress): MikuBuilder[F] =
+    copy(socketAddress = socketAddress)
+
+  def withExecutionContext(executionContext: ExecutionContext): MikuBuilder[F] =
+    copy(ec = executionContext)
+
+  def withServiceErrorHandler(serviceErrorHandler: ServiceErrorHandler[F]): MikuBuilder[F] =
+    copy(serviceErrorHandler = serviceErrorHandler)
+
+  //Todo: Router-hack thingy
+  def mountService(service: HttpService[F], prefix: String): MikuBuilder[F] =
+    copy(httpService = service)
+
+  def start: F[Server[F]] = F.delay {
+    val eventLoop: MultithreadEventLoopGroup = transport match {
+      case Jdk => new NioEventLoopGroup()
+      case Native =>
+        if (Epoll.isAvailable)
+          new EpollEventLoopGroup()
+        else {
+          logger.info("Falling back to NIO EventLoopGroup")
+          new NioEventLoopGroup()
+        }
+    }
+
+    val allChannels: DefaultChannelGroup = new DefaultChannelGroup(eventLoop.next())
+
+    val (_, boundAddress) = startNetty(eventLoop, allChannels)
+    val server = new Server[F] {
+      def shutdown: F[Unit] = F.delay {
+        // First, close all opened sockets
+        allChannels.close().awaitUninterruptibly()
+        // Now shutdown the event loop
+        eventLoop.shutdownGracefully()
+
+        logger.info("Shut down gracefully :)")
```

Comment:
Maybe we could add the address that's shut down here, in case we have multiple servers running.